### PR TITLE
[14.0][FIX] purchase_discount: ocb compatibility

### PR DESCRIPTION
**Pull request by chienandalu**
Depends on:

- [x] https://github.com/OCA/OCB/pull/1193

Since oca/ocb@21a01c303347b8b80d0ffa555ed72e680c251e76 we already have proper hooks in OCB so we don't need to do nasty hacks, but we need to keep compatibility with both Odoo and OCB.

cc @Tecnativa TT45385

please review @pedrobaeza @victoralmau

### CODE REVIEW

**Review by pedrobaeza — APPROVED** [review]

File: purchase_discount/models/purchase_order.py [hunk 1]
```diff
@@ -65,6 +65,8 @@ def _get_stock_move_price_unit(self):
         HACK: This is needed while https://github.com/odoo/odoo/pull/29983
         is not merged.
         """
+        if hasattr(self.env, "ocb"):
```
Comment (legalsylvain) [inline purchase_discount/models/purchase_order.py:68]: naive question : I didn't know ocb adds an extra val in self.env. is it documented somewhere ?

**Review by legalsylvain — APPROVED** [review]
code review / no test.

File: purchase_discount/models/purchase_order.py (hunk 1 above)
Comment (pedrobaeza) [reply purchase_discount/models/purchase_order.py:68]: Check OCA/OCB#1193, which is flagged on the header :wink:

Comment (legalsylvain) [reply purchase_discount/models/purchase_order.py:68]: thanks !

**Review by victoralmau — APPROVED** [review]

**Comment by chienandalu** [comment]
Is there a way to force a new OCB package for the CI build? (https://github.com/OCA/oca-ci/pkgs/container/oca-ci%2Fpy3.6-ocb14.0)

**Comment by pedrobaeza** [comment]
We have to wait till tomorrow...

**Comment by pedrobaeza** [comment]
@chienandalu why is it still failing?

**Comment by chienandalu** [comment]
> @chienandalu why is it still failing?

The package was updated 1 hour ago. Now is green :)

**Pull request merged**